CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+cmake_minimum_required(VERSION 3.16)
+
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+option(SANITIZE_ADDRESS "Address Sanitizer." OFF)
+
+if (SANITIZE_ADDRESS)
+set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
+set (CMAKE_LINKER_FLAGS_DEBUG "${CMAKE_LINKER_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
+endif()
+
+include(FetchContent)
+
+if(NOT InstallVendor)
+function(InstallVendor project_name git_repo branch)
+        IF (NOT ${project_name})
+                FetchContent_Declare(
+                        ${project_name}
+                        GIT_REPOSITORY ${git_repo}
+                        GIT_TAG ${branch}
+                        SOURCE_DIR "${PROJECT_SOURCE_DIR}/vendor/${project_name}"
+                        BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/${project_name}"
+                        SUBBUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}/${project_name}-sub"
+                )
+                FetchContent_MakeAvailable(${project_name})
+                set(${project_name} CACHE INTERNAL "")
+                message(STATUS "Found ${project_name} @ ${branch}")
+        endif()
+endfunction()
+endif()
+
+set(This oof)
```

Comment:
Come on, this is completely unnecessary and it's just more characters than spelling `oof` out directly.

Reply:
I will not agree with you on that as it is seen in pretty much every CMake file and makes things easy and consistent. Should the CMake file be in here in makes things easier.
I'll leave the end decision to @s9w unless you would agree.

Reply:
> seen in pretty much every CMake file

Pretty much every CMake script is composed of oodles of spaghetti for all the wrong reasons. Let's not copy that if we can.